### PR TITLE
fix(TDI-40461): Implement extended Excel for fileio

### PR DESCRIPTION
**Pull request by wwang-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-40461

**What is the new behavior?**
https://jira.talendforge.org/browse/TDI-40461

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/46/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/46/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTDI-40461)

File: components/components-fileio/simplefileio-definition/src/main/java/org/talend/components/simplefileio/local/EncodingType.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.talend.components.simplefileio.local;
+
+public enum EncodingType {
```
Comment (jzhao-talend) [inline components/components-fileio/simplefileio-definition/src/main/java/org/talend/components/simplefileio/local/EncodingType.java]: can't reuse `org.talend.components.common.EncodingTypeProperties` ?

File: components/components-fileio/simplefileio-runtime/pom.xml [hunk 2]
```diff
@@ -213,6 +217,111 @@
...
+        -->
+        <dependency>
+    		<groupId>org.apache.tika</groupId>
+		    <artifactId>tika-parsers</artifactId>
+	    	<version>${tika.version}</version>
+			<exclusions>
+                <exclusion>
+                    <groupId>org.osgi</groupId>
+                    <artifactId>org.osgi.core</artifactId>
+                </exclusion>
```
Comment (jzhao-talend) [inline components/components-fileio/simplefileio-runtime/pom.xml]: Don't forget format code. ^_^

File: components/components-fileio/simplefileio-definition/src/main/java/org/talend/components/simplefileio/local/EncodingType.java (hunk 1 above)
Comment (wwang-talend) [reply components/components-fileio/simplefileio-definition/src/main/java/org/talend/components/simplefileio/local/EncodingType.java]: not sure Widget.ENUMERATION_WIDGET_TYPE can works for data-catalog, so write it simplely like this.

File: components/components-fileio/simplefileio-runtime/pom.xml (hunk 2 above)
Comment (wwang-talend) [reply components/components-fileio/simplefileio-runtime/pom.xml]: will format it before integrate it, if format now, a big changeset.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/75/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/75/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTDI-40461)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/79/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/79/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTDI-40461)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/81/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/81/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTDI-40461)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/83/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/83/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTDI-40461)

**Pull request merged**